heidegger_index/views.py

```diff
@@ -40,3 +43,36 @@ class LemmaDetailView(DetailView):
     model = Lemma
     template_name = "lemma_detail.html"
     context_object_name = "lemma"
+
+    # Just a basic copy of index.py find_ref
```

Comment:
Misschien kun je dit abstraheren in een aparte functie (die je bijvoorbeeld in `heidegger_index.utils` kwijt kan) en die in `index.py` en hier aanroepen (DRY!)

Reply:
Ik stuit hier echter wel op een probleem, zoals de volgende error laat zien:
```shell
django.core.exceptions.ImproperlyConfigured: Requested setting INDEX_FILE, 
but settings are not configured. You must either define the environment 
variable DJANGO_SETTINGS_MODULE or call settings.configure() before 
accessing settings.
```
Het probleem is dat [`index.py`](https://github.com/andredelft/heidegger-index/blob/main/index/index.py) een functie uit `utils.py` importeert, maar niet binnen een django-environment. [Zie ook django's docs.](https://docs.djangoproject.com/en/4.0/topics/settings/#calling-django-setup-is-required-for-standalone-django-usage)

Zou jij hier django's advies volgen of de path naar `heidegger-index.yml` gewoon hardcoden in `utils.py`?

Reply:
Ah, ik had even wat beter moeten kijken, het compliceert de zaak inderdaad als er variabelen uit `settings.py` gebruikt worden...

Wat je misschien het beste kunt doen is de hele index als variabele meegeven. Dus:

```python
def match_lemmata(search_term, index, max_l_dist=2, num_results=5, include_search_term=True):

      key_to_lemma = {
          gen_sort_key(lemma): lemma for lemma in index.keys()
      }
      keys = key_to_lemma.keys()

      # etc...
```

En dan bij het aanroepen op de respectievelijke plekken (even globaal):

```python
with open('path/to/index.yml') as f:
    index = yaml.load(f)

matches = match_lemmata(search_term, index)
```